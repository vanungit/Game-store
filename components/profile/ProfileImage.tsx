import React, { FC, useState } from "react";
import ReactFileReader from "react-file-reader";
import style from "./profileImage.module.scss";
import mainStyle from "../../styles/main.module.css";
import { IUserProfile } from "../../types/UserProfile";

interface IProfileImage extends IUserProfile {}

const ProfileImage: FC<IProfileImage> = ({ avatar }) => {
  const defaultAvatarUrl = "https://winnote.ru/wp-content/uploads/2016/01/1454222417_del_recent_avatar1.png";
  const isImg = avatar || defaultAvatarUrl;
  const [image, setImage] = useState<string>(isImg);

  const handleFiles = (files) => {
    setImage(files.base64);
  };

  return (
    <div>
      <img className={style.image} src={image} alt="" />
      <ReactFileReader base64 handleFiles={handleFiles}>
        <button className={mainStyle.btn} type="button">
          Change profile image
        </button>
      </ReactFileReader>
    </div>
  );
};

export default ProfileImage;
