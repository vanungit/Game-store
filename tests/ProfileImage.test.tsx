import React from "react";
import renderer from "react-test-renderer";
import ProfileImage from "../components/profile/ProfileImage";

describe("ProfileInfo component tests", () => {
  it("should render the default avatar", () => {
    const component = renderer.create(<ProfileImage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
