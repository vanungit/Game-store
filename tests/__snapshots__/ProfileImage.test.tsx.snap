// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo component should render the default avatar 1`] = `
<div>
  <img
    alt=""
    className="image"
    src="https://winnote.ru/wp-content/uploads/2016/01/1454222417_del_recent_avatar1.png"
  />
  <div
    className="react-file-reader"
  >
    <input
      accept="image/*"
      className="react-file-reader-input"
      disabled={false}
      id="3ff02448-708d-43e0-b007-77d7e0e724a1"
      multiple={false}
      onChange={[Function]}
      style={
        Object {
          "opacity": "0",
          "position": "fixed",
          "width": "0px",
        }
      }
      type="file"
    />
    <div
      className="react-file-reader-button"
      onClick={[Function]}
    >
      <button
        className="btn"
        type="button"
      >
        Change profile image
      </button>
    </div>
  </div>
</div>
`;
